packages/core/types/@chainlink/index.d.ts

```diff
@@ -102,6 +102,48 @@ declare module '@chainlink/types' {
     warning?: string
   }
 
+
+  export type MakeWSHandler = () => WSSubscriptionHandler
+
+  export interface WSSubscriptionHandler {
+    /**
+     * Connection information
+     */
+    connection: {
+      url: string
+      protocol?: any
+    }
+    /**
+     * Gets the subscription message necessary to subscribe to the feed channel
+     */
+    subscribe: (input: AdapterRequest) => any
+    /**
+     * Gets the unsubscription message necessary to unsubscribe to the feed channel
+     */
+    unsubscribe: (input: any) => any
+    /**
+     * Gets the response from the incoming message
+     */
+    parse: (message: any) => number | string
+    /**
+     * Filter any message that is not from a subscribed channel
+     */
+    filter: (message: any) => boolean
+    /**
+     * Determines if the incoming message is an error
+     */
+    isError: (message: any) => boolean
+    /**
+     * Based on the incoming message, returns its corresponding subscription message
+     */
+    subsFromMessage: (message: any) => any
+    /**
+     * Optional. If defined, the response will be formatted using this function
+     */
+    toAdapterResponse?: (message: any) => any
```

Comment:
Why do we need both `parse` and `toAdapterResponse`?

If there is not something I'm missing, I suggest we use just the latter:

```suggestion
    toResponse: (message: any) => AdapterResponse
```

Reply:
`parse` get the message relevant information, `toAdapterResponse` is how the response will be saved in cache. It's possible that some adapter wants an specific response format

Reply:
But why would the WS component need the "message relevant information". This can all be abstracted by our `AdaperResponse` type that adapters would map messages to.

Reply:
Yep I see